libraries/bsg_manycore.cpp

```diff
@@ -120,53 +120,14 @@ static int hb_mc_manycore_packet_rx_internal(hb_mc_manycore_t *mc,
 // FIFO Helper Functions //
 ///////////////////////////
 
-static int hb_mc_manycore_fifo_get_isr_bit(hb_mc_manycore_t *mc, hb_mc_direction_t dir, uint32_t bit, uint32_t *v)
-{
-        uintptr_t isr_addr = hb_mc_mmio_fifo_get_reg_addr(dir, HB_MC_MMIO_FIFO_ISR_OFFSET);
-        uint32_t tmp;
-        int err;
-
-        err = hb_mc_manycore_mmio_read32(mc, isr_addr, &tmp);
-        if (err != HB_MC_SUCCESS) {
-                manycore_pr_err(mc, "Failed to read bit in %s ISR register: %s\n",
-                                hb_mc_direction_to_string(dir), hb_mc_strerror(err));
-                return err;
-        }
-
-        *v = (tmp >> bit) & 1;
-        return HB_MC_SUCCESS;
-}
-
-static int hb_mc_manycore_fifo_clear_isr_bit(hb_mc_manycore_t *mc, hb_mc_direction_t dir, uint32_t bit)
-{
-        uintptr_t isr_addr = hb_mc_mmio_fifo_get_reg_addr(dir, HB_MC_MMIO_FIFO_ISR_OFFSET);
-        int err;
-
-        err = hb_mc_manycore_mmio_write32(mc, isr_addr, (1<<bit));
-        if (err != HB_MC_SUCCESS) {
-                manycore_pr_err(mc, "Failed to set bit in %s ISR register: %s\n",
-                                hb_mc_direction_to_string(dir), hb_mc_strerror(err));
-                return err;
-        }
-
-        return HB_MC_SUCCESS;
-}
-
 static int hb_mc_manycore_tx_fifo_get_vacancy(hb_mc_manycore_t *mc,
                                               hb_mc_fifo_tx_t type,
                                               uint32_t *vacancy)
 {
         const char *typestr = hb_mc_fifo_tx_to_string(type);
-        uintptr_t vacancy_addr = hb_mc_mmio_fifo_get_reg_addr(type, HB_MC_MMIO_FIFO_TX_VACANCY_OFFSET);
-        int err;
-
-        err = hb_mc_manycore_mmio_read32(mc, vacancy_addr, vacancy);
-        if (err != HB_MC_SUCCESS) {
-                manycore_pr_err(mc, "Failed to get %s vacancy\n", typestr);
-                return err;
-        }
-
-        return HB_MC_SUCCESS;
+        manycore_pr_dbg(mc, "We do not check the %s FIFO's vacancy\n", typestr);
+        vacancy = NULL;
+        return HB_MC_FAIL;
 }
```

Comment:
Why not? It seems like poor form to return HB_MC_FAIL in all cases. If you don't expect this method to be used anymore it should be removed